### ISSUE TITLE
unknown file type error when registering records

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#1145]
Upon attempting to register records and create a new CO data group, an 'unknown file type' error was seen.

.csv that was being uploaded to register records was: 
[https://app.zenhub.com/files/99837877/3ee4f34d-6f74-4f3c-9a2b-9ac1ab52bf2d/download](https://app.zenhub.com/files/99837877/3ee4f34d-6f74-4f3c-9a2b-9ac1ab52bf2d/download)

Also, the data group was still created but did not have any registered records: http://factotum.epa.gov/datagroup/550/

Definition of Done:
1. Resolve bug of unknown file type error
2. If error when registering records (as above), data group should not be created (I thought this was resolved in #659?)
3. Also see #1123; test for #659 needs to be included in this ticket

**Comment by mcovalt** [comment]
See also https://github.com/HumanExposure/factotum/issues/1104

**Comment by dmlyons2** [comment]
@kdionisio is going to put in a new ticket for an AJAX interface for uploads.

**Comment by zachmunzenrider** [comment]
I've located the culprit in the "unknown file type" error ticket. This error gets thrown when the user tries to load a particularly large (in bytes) csv file. Here's the code:
```        if type(f.file) is io.StringIO:
            self.f = f
        elif type(f.file) is io.BytesIO:
            self.f = io.TextIOWrapper(f.file, encoding="utf-8-sig", newline="")
        else:
            raise ValueError("Unknown file type.")```
What happens, as per https://docs.djangoproject.com/en/2.2/topics/http/file-uploads/#upload-handlers,  is that a file above the FILE_UPLOAD_MAX_MEMORY_SIZE threshold gets stored in a temporary file, at which point it ceases to be of type io.BytesIO and becomes instead a django.core.files.temp.TemporaryFile Thus it throws that ValueError.

So now I'm trying to decide what to do about it. I've started by explicitly setting a value for  FILE_UPLOAD_MAX_MEMORY_SIZE that's bigger than the uploaded .csv, and that's now been chugging along on my computer for the past 5 minutes or so.
